File: apps/api/app/api/routes/health.py
"""Health check route — the one endpoint that skips auth (PUBLIC_PATHS in
core/middleware.py). No service, no repository."""

from datetime import datetime, timezone

from fastapi import APIRouter

from app.schemas import HealthResponse

router = APIRouter()


@router.get("/api/health", response_model=HealthResponse)
def health() -> HealthResponse:
    """Report that the API process is alive, with a UTC timestamp."""
    return HealthResponse(
        status="ok",
        service="note2action-api",
        time=datetime.now(timezone.utc).isoformat(),
    )

File: packages/shared/src/health.ts
/** Zod schema + type for the health check, parsed by apps/web's health.queries.ts. */
import { z } from "zod";

/** Shape of the API's health-check response. */
export const HealthResponse = z.object({
  status: z.string(),
  service: z.string(),
  /** ISO-8601 timestamp. */
  time: z.string(),
});
export type HealthResponse = z.infer<typeof HealthResponse>;
